Makefile

```diff
@@ -211,21 +205,36 @@ debug.analyzer:
 	sed -e 's/type Flow struct {/type Flow struct { XXX_state flowState `json:"-"`/' -i.bak flow/flow.pb.go
 	gofmt -s -w flow/flow.pb.go
 
-.PHONY: .easyjson.all
-.easyjson.all: builddep .proto ${EASYJSON_FILES_ALL}
-	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson -all ${EASYJSON_FILES_ALL}
+.proto: govendor flow/flow.pb.go filters/filters.pb.go http/wsstructmessage.pb.go
+
+.PHONY: .proto.clean
+.proto.clean:
+	find . \( -name *.pb.go ! -path './vendor/*' \) -exec rm {} \;
+
+GEN_EASYJSON_FILES_TAG := $(patsubst %.go,%_easyjson.go,$(EASYJSON_FILES_TAG))
+GEN_EASYJSON_FILES_TAG_LINUX := $(patsubst %.go,%_easyjson.go,$(EASYJSON_FILES_TAG_LINUX))
+GEN_EASYJSON_FILES_TAG_OPENCONTRAIL := $(patsubst %.go,%_easyjson.go,$(EASYJSON_FILES_TAG_OPENCONTRAIL))
+
+%_easyjson.go: %.go
+	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson $<
 
-.PHONY: .easyjson.tag
-.easyjson.tag: builddep ${EASYJSON_FILES_TAG}
-	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson ${EASYJSON_FILES_TAG}
+flow/flow.pb_easyjson.go: flow/flow.pb.go
+	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson -all $<
 
-.PHONY: .easyjson.tag.linux
-.easyjson.tag.linux: builddep ${EASYJSON_FILES_TAG_LINUX}
-	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson -build_tags linux ${EASYJSON_FILES_TAG_LINUX}
+topology/probes/netlink/netlink_easyjson.go: topology/probes/netlink/netlink.go
+	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson -build_tags linux $<
 
-.PHONY: .easyjson.tag.opencontrail
-.easyjson.tag.opencontrail: builddep ${EASYJSON_FILES_TAG_OPENCONTRAIL}
-	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson -build_tags "opencontrail" ${EASYJSON_FILES_TAG_OPENCONTRAIL}
+topology/probes/socketinfo/connection_easyjson.go: topology/probes/socketinfo/connection.go
+	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson -build_tags linux $<
+
+topology/probes/opencontrail/routing_table_easyjson.go: $(EASYJSON_FILES_TAG_OPENCONTRAIL)
+	$(call VENDOR_RUN,${EASYJSON_GITHUB}) easyjson -build_tags opencontrail $<
+
+.easyjson: flow/flow.pb_easyjson.go $(GEN_EASYJSON_FILES_TAG) $(GEN_EASYJSON_FILES_TAG_LINUX) $(GEN_EASYJSON_FILES_TAG_OPENCONTRAIL)
```

Comment:
But in the case when `WITH_OPENCONTRAIL=false` I don't think you should build `routing_table_easyjson.go`

Reply:
That brings other problems: when we generate the tarball (for the release, the packaging, ...), we also need to put WITH_OPENCONTRAIL=true. And I'm sure we'll forget to do so.

So I prefer to stick with this approach

Reply:
fine